azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -546,6 +556,30 @@ def __try_set_cpu_quota(quota):  # pylint: disable=unused-private-member
                 return False
             return True
 
+        def _check_processes_in_agent_cgroup_before_enable(self, agent_slice):
+            """
+            This check ensures that before we enable the agent's cgroups, there are no unexpected processes in the agent's cgroup already.
+
+            The issue we observed that long running extension processes may be in agent cgroups if agent goes this cycle enabled(1)->disabled(2)->enabled(3).
+            1. Agent cgroups enabled in some version
+            2. Disabled agent cgroups due to check_cgroups regular check, now extension runs will be in agent cgroups.
```

Comment:
I don't understand #2. 
If cgroups are disabled because some unexpected process is found in agent cgroup, then why would extension runs now be in agent cgroups?

Reply:
Once agent cgroups disabled (disable for both agent and extensions), we don't run the extensions in the extension slice. That means it will be in agent cgroups.

Reply:
could you please add those details to the comment